shell/common/animator.cc

```diff
@@ -74,13 +76,27 @@ void Animator::SetDimensionChangePending() {
   dimension_change_pending_ = true;
 }
 
-void Animator::EnqueueTraceFlowId(uint64_t trace_flow_id) {
+void Animator::EnqueueTraceData(const PointerDataPacket& packet,
+                                uint64_t trace_flow_id) {
+  std::vector<PointerData> pointer_datas;
+  constexpr size_t kBytesPerPointerData =
+      kPointerDataFieldCount * kBytesPerField;
+  const auto& buffer = packet.data();
+  size_t buffer_length = buffer.size();
+  for (size_t i = 0; i < buffer_length / kBytesPerPointerData; i++) {
+    PointerData data;
+    memcpy(&data, &buffer[i * kBytesPerPointerData], sizeof(PointerData));
+    pointer_datas.push_back(data);
+  }
   fml::TaskRunner::RunNowOrPostTask(
       task_runners_.GetUITaskRunner(),
-      [self = weak_factory_.GetWeakPtr(), trace_flow_id] {
+      [self = weak_factory_.GetWeakPtr(), pointer_datas, trace_flow_id] {
```

Comment:
Instead of manually copying `pointer_datas`, it seems to be easier to
1. Add function `GetPointerData(size_t i)` to `PointerDataPacket`
2. Directly copy the whole `PointerDataPacket` in the lambda, and then `pointer_datas_.push_back(packet.GetPointerData(i))` inside.